src/plugins/content_management/demo/todo/todos.tsx

```diff
@@ -0,0 +1,121 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React from 'react';
+import { EuiButtonIcon, EuiCheckbox, EuiFieldText, EuiButtonGroup, EuiSpacer } from '@elastic/eui';
+
+import {
+  useSearchContentQuery,
+  useCreateContentMutation,
+  useDeleteContentMutation,
+  useUpdateContentMutation,
+  // eslint-disable-next-line @kbn/imports/no_boundary_crossing
+} from '../../public/content_client';
+import type { Todo } from './todos_client';
+import { SearchIn, SearchOut, CreateIn, DeleteIn, UpdateIn } from '../../common';
+
+export const Todos = () => {
+  const [filterIdSelected, setFilterIdSelected] = React.useState('all');
+
+  const { data, isLoading, isError, error } = useSearchContentQuery<SearchIn, SearchOut<Todo>>({
+    contentType: 'todos',
+    params: {
+      filter: filterIdSelected,
+    },
+  });
+
+  const addTodoMutation = useCreateContentMutation<CreateIn<'todos', { title: string }, Todo>>();
+  const deleteTodoMutation = useDeleteContentMutation<DeleteIn<'todos', { id: string }, Todo>>();
+  const updateTodoMutation = useUpdateContentMutation<UpdateIn<'todos', Todo, Todo>>();
+
+  const filters = [
+    {
+      id: `all`,
+      label: 'All',
+    },
+    {
+      id: `completed`,
+      label: 'Completed',
+    },
+    {
+      id: `todo`,
+      label: 'Todo',
+    },
+  ];
+
+  if (isLoading) return <p>Loading...</p>;
+  if (isError) return <p>Error: {error}</p>;
+
+  return (
+    <>
+      <EuiButtonGroup
+        legend="Todo filters"
+        options={filters}
+        idSelected={filterIdSelected}
+        onChange={(id) => setFilterIdSelected(id)}
+      />
+      <EuiSpacer />
+      <ul>
+        {data.hits.map((todo: Todo) => (
+          <React.Fragment key={todo.id}>
+            <li style={{ display: 'flex', alignItems: 'center' }}>
+              <EuiCheckbox
+                id={todo.id + ''}
+                key={todo.id}
+                checked={todo.completed}
+                onChange={(e) => {
+                  updateTodoMutation.mutate({
+                    contentType: 'todos',
+                    data: {
+                      ...todo,
+                      completed: e.target.checked,
+                    },
+                  });
+                }}
+                label={todo.title}
+                data-test-subj={`todoCheckbox-${todo.id}`}
+              />
+
+              <EuiButtonIcon
+                style={{ marginLeft: '8px' }}
+                display="base"
+                iconType="trash"
+                aria-label="Delete"
+                color="danger"
+                onClick={() => {
+                  deleteTodoMutation.mutate({ contentType: 'todos', data: { id: todo.id } });
```

Comment:
Why do I have to pass `contentType: 'todos'` everywhere where working with a single type? 
Should we offer all apis as both "global" and "scoped" versions? "global" will require `contentType` specified in each usage and scope will require it only one. Can all other type be bound together with the scoped version so we don't have specify them with each usage like now?

Reply:
Should we add a set of alternative apis which is "scoped" to a contentType? 


![](https://user-images.githubusercontent.com/7784120/219419916-1770dfa6-54bd-451f-81db-32389f75a5ad.png)